File: TransCoder_tokenized_test_set_functions/python/SEARCH_AN_ELEMENT_IN_A_SORTED_AND_PIVOTED_ARRAY.py
def search ( arr, l, h, key ) :
    if l > h :
        return - 1
    mid = ( l + h ) //2
    ifarr[mid]==key:
        returnmid
    ifarr[l]<=arr[mid]:
        ifkey>=arr[l]andkey<=arr[mid]:
            returnsearch(arr,l,mid-1,key)
        returnsearch(arr,mid+1,h,key)
    ifkey>=arr[mid]andkey<=arr[h]:
        returnsearch(a,mid+1,h,key)
    returnsearch(arr,l,mid-1,key)

File: results/py/SEARCH_AN_ELEMENT_IN_A_SORTED_AND_PIVOTED_ARRAY.cpp
int f_filled ( int arr [ ] , int l , int h , int key ) {
  if ( l > h ) return - 1 ;
  int mid = ( l + h ) / 2 ;
  if ( ifarr [ mid ] == key ) returnmid ++ ;
  if ( ifarr [ l ] <= arr [ mid ] ) {
    if ( ifkey >= arr [ l ] && andkey <= arr [ mid ] ) returnf_filled ( arr , l , mid - 1 , key ) ;
    returnf_filled ( arr , mid + 1 , h , key ) ;
  }
  if ( ifkey >= arr [ mid ] && andkey <= arr [ h ] ) returnf_filled ( a , mid + 1 , h , key ) ;
  returnf_filled ( arr , l , mid - 1 , key ) ;
  return mid ;
}
